graph.py

```diff
@@ -115,85 +114,94 @@ def incidence(self) -> List["Edge"]:
 
     @property
     def neighbours(self) -> ["Vertex"]:
-        """
-        Returns the list of neighbors of the vertex.
-        """
+        """Get the list of neighbours of the vertex."""
+
         return list(self._incidence.keys())
 
     @property
     def degree(self) -> int:
-        """
-        Returns the degree of the vertex
-        """
+        """Get the degree of the vertex."""
+
         return sum(map(len, self._incidence.values()))
 
+    @property
+    def label(self):
+        return self._label
+
+    @label.setter
+    def label(self, label):
+        self._label = label
+
 
 class Edge(object):
-    """
-    Edges have properties `tail` and `head` which point to the end vertices
-    (`Vertex` objects). The order of these matters when the graph is directed.
-    """
+    """An edge has a tail and a head which point to the end vertices. The order of these matters if the graph is
+    directed."""
 
     def __init__(self, tail: Vertex, head: Vertex, weight=None):
-        """
-        Creates an edge between vertices `tail` and `head`
-        :param tail: In case the graph is directed, this is the tail of the arrow.
-        :param head: In case the graph is directed, this is the head of the arrow.
+        """Create an edge between vertices `tail` and `head`.
+
+        :param Vertex tail: In case the graph is directed, this is the tail of the arrow.
+        :param Vertex head: In case the graph is directed, this is the head of the arrow.
         :param weight: Optional weight of the vertex, which can be any type, but usually is a number.
-        :param graph: Optional graph the
         """
-        if set(tail.graphs) != set(head.graphs):
+
+        if len(set(tail.graphs).intersection(set(head.graphs))) == 0:
```

Comment:
Is dit sneller?

Reply:
Het doet echter het volgende: de intersectie van grafen van de twee vertices moet niet leeg zijn. In andere woorden: er moet een graaf bestaan waar beide vertices in bestaan. Anders kun je geen edge tussen hen maken.

Deze edge wil je namelijk later toevoegen aan die graaf. Het doet er niet toe of deze vertices allebei in al hun grafen voorkomen, enkel in de graaf waar deze edge aan toegevoegd worden.

Dit is echt matig design van de `Graph-Vertex-Edge` klassen die we hebben gekregen, want al die stomme checks zijn compleet overbodig. 

/rant Ware het niet dat het teveel tijd zou kosten, dan zou ik die drie klassen volledig verwijderen en herbouwen in een veel primitievere en eenvoudigere variant.

Reply:
Ik snap wat het doet;) Vroeg me vooral af of intersectie + length berekenen sneller was dan het bovenstaande.

Reply:
Ah, op die manier. Het is denk ik langzamer, maar het andere is incorrect. De tests gaan overigens wel goed met `if set(tail.graphs) != set(head.graphs):`. Dat was niet zo in een tussentijdse versie.

Wat mij betreft kan de hele check eruit. Als jij een edge maakt tussen twee willekeurige vertices: prima toch? De vertices worden toch wel aan de graaf toegevoegd in `Graph#add_edge`, indien nodig.